Validation errors when using Node 7.0.0 with PM2

I'm suffering with this bug for two days now:
Just upgreded to Node 7.0.0, and when i started to host with PM2 (single instance with fork, PM2 2.1.4, Mongoose 4.6.6) and gtting these error messages after the application starts:

`TypeError: Cannot convert undefined or null to object
    at Function.keys ()
    at ValidationError.toString (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mongoose/lib/error/validation.js:49:10)
    at Function.prepareStackTrace (/usr/lib/node_modules/pm2/node_modules/source-map-support/source-map-support.js:365:16)
    at Function.captureStackTrace ()
    at new MongooseError (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mongoose/lib/error.js:12:11)
    at ValidationError (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mongoose/lib/error/validation.js:17:19)
    at model.Document.invalidate (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mongoose/lib/document.js:1462:32)
    at _init (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mongoose/lib/document.js:354:18)
    at init (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mongoose/lib/document.js:330:7)
    at model.Document.init (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mongoose/lib/document.js:295:3)
    at completeOne (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mongoose/lib/query.js:1609:10)
    at Immediate. (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mongoose/lib/query.js:1271:13)
    at Immediate. (/newhope/node_projects/BlitzInfoServerPostThesis/node_modules/mquery/lib/utils.js:137:16)
    at runCallback (timers.js:637:20)
    at tryOnImmediate (timers.js:610:5)
    at processImmediate [as _immediateCallback] (timers.js:582:5)
[4:46:19 pm][PM2] App [BlitzInfoPROD] with id [0] and pid [12694], exited with code [1] via signal [SIGINT]`

And i'm also expirencing strange CPU usages.

I'm pretty sure my schemas are valid, because it's not an issue when i'm running it directly from the command line.
The only solution for me to make it work with PM2 is downgrading Node to version 6.9.1.

Looking at the code, it appears to be a symptom of Error.captureStackTrace and using source-map-support.  When I disable source-map-support everything starts working again.  Additionally though, mongoose attempts to pass a null/undefined reference to Object.keys in the ValidationError.toString method which is ultimately what is causing the issue.

--- a/lib/error/validation.js
+++ b/lib/error/validation.js
@@ -46,7 +46,7 @@ ValidationError.prototype.toString = function() {
   var ret = this.name + ': ';
   var msgs = [];
 
-  Object.keys(this.errors).forEach(function(key) {
+  Object.keys(this.errors || {}).forEach(function(key) {
     if (this === this.errors[key]) {
       return;
     }

